### PR TITLE
[DO NOT SQUASH] Warnings cleanup

### PR DESCRIPTION
**Pull request by krzysz00**
This one's exactly what it says it is.

`skip-ci` has been applied so that we can get to merging it after some of the more substantial stuff goes in - we only have so many CI nodes at the moment, after all.

### CODE REVIEW

File: mlir/lib/CAPI/Dialect/MIGraphX.cpp [hunk 1]
```diff
@@ -57,7 +57,7 @@ void mlirGetKernelInfo(MlirModule module, int *size, void *data) {
     for (int i = 0; i < argSize; i++)
       argData[i + 1] = info[i];
     char *nameData = (char *)(argData + argSize + 1);
-    for (int i = 0; i < kernelName.size(); i++) {
```
Comment (jungpark-mlir) [inline mlir/lib/CAPI/Dialect/MIGraphX.cpp:60]: I suppose size() only returns constant value and compiler can optimize this. good to have the correct type, thanks.

File: mlir/test/CAPI/mixr_cobj.cpp [hunk 2]
```diff
@@ -202,11 +200,11 @@ static bool constructAndTraverseIr(MlirContext ctx) {
                        mlir::PassManager::Nesting::Implicit);
 
   mlir::migraphx::addHighLevelPipeline(pm);
-  pm.run(module);
+  (void)pm.run(module);
```
Comment (jungpark-mlir) [inline mlir/test/CAPI/mixr_cobj.cpp:203]: why do we need this?

File: mlir/lib/CAPI/Dialect/MIGraphX.cpp (hunk 1 above)
Comment (krzysz00) [reply mlir/lib/CAPI/Dialect/MIGraphX.cpp:60]: Yeah. I've seen the `i = 0, e = ...; i < e` style in a bunch of LLVM code and I'm assuming there's a reason for it

File: mlir/test/CAPI/mixr_cobj.cpp (hunk 2 above)
Comment (krzysz00) [reply mlir/test/CAPI/mixr_cobj.cpp:203]: `-Wunused-result` is the setting, IIRC. `LogicalResult` has a NODISCARD on it, so the compiler will yell if you don't check it or explicitly point out you're not checking it with something like that `(void)`

Comment (jungpark-mlir) [reply mlir/test/CAPI/mixr_cobj.cpp:203]: Right, i'd have better consumed it. (void) should be fine for now.

**Review by jungpark-mlir — APPROVED** [review]

**Pull request merged**